src/TypeOps.hs

```diff
@@ -46,3 +47,12 @@ import           Control.Type.Operator (type ($), type (<+>))
 type family Each (c :: [k -> Constraint]) (as :: [k]) where
     Each c '[] = (() :: Constraint)
     Each c (h ': t) = (c <+> h, Each c t)
+
+-- | Map several constraints over a single variable.
+--
+-- @
+-- a :: With [Show, Read] a => a -> a
+-- =
+-- a :: (Show a, Read a) => a -> a
+-- @
+type With a b = a <+> b
```

Comment:
I think it's better to define `With` in terms of `Each` because `Each` seems more natural than some unknown and weird operator. Also, documentation of `Each` is in this module with examples, so it's easier to understand how `With` works if it uses `Each`.